python/beeai_framework/emitter/emitter.py

```diff
@@ -162,7 +164,7 @@ def create_matcher() -> MatcherFn:
             if not match_nested:
 
                 def match_same_run(event: EventMeta) -> bool:
-                    return self.trace is None or self.trace.run_id == event.trace.run_id
+                    return self.trace is None or event.trace is None or self.trace.run_id == event.trace.run_id
```

Comment:
this does not seem right

Reply:
I just checked the TS impl and you are correct, in the `event.trace is None` case this should be False, not True